modules/mono/editor/script_class_parser.cpp

```diff
@@ -312,27 +314,109 @@ Error ScriptClassParser::_parse_class_base(Vector<String> &r_base) {
 
 	Token tk = get_token();
 
+	bool generic = false;
 	if (tk == TK_OP_LESS) {
-		// We don't add it to the base list if it's generic
 		Error err = _skip_generic_type_params();
 		if (err)
 			return err;
-	} else if (tk == TK_COMMA) {
-		Error err = _parse_class_base(r_base);
+		// We don't add it to the base list if it's generic
+		generic = true;
+		tk = get_token();
+	}
+
+	if (tk == TK_COMMA) {
+		Error err = _parse_class_base(r_base, false);
 		if (err)
 			return err;
-		r_base.push_back(name);
+	} else if (tk == TK_IDENTIFIER && String(value) == "where") {
+		Error err = _parse_type_constraints();
+		if (err) {
+			return err;
+		}
+
+		// An open curly bracket was parsed by _parse_type_constraints, so we can exit
 	} else if (tk == TK_CURLY_BRACKET_OPEN) {
-		r_base.push_back(name);
+		// we are finished when we hit the open curly bracket
 	} else {
 		error_str = "Unexpected token: " + get_token_name(tk);
 		error = true;
 		return ERR_PARSE_ERROR;
 	}
 
+	// only the first inherited entry can be a base class
+	if (first && !generic) {
+		r_base.push_back(name);
+	}
```

Comment:
Like I said, `base` is not meant to be the base class but everything the class implements and explicitly extends. It's not like it changes anything thought.

Reply:
works for me!